### DISCUSSION TITLE
Problems processing with stackStripMap.py (ISCE2)

### DISCUSSION THREAD

**Discussion by fgdiazm** [isce-framework/isce2#365] [Q&A]
Hi everyone,

I am trying to process a group of TerraSARX images with stackStripMap.py, I managed to generate the SLCs with unpackFrame_TSX.py.

I am trying to apply stackStripMap.py with the following option

 stackStripMap.py -s / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / -d / home / work / geored-insar / Isce2 /Cartagena/DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -x '10 .06 10.36 -75.62 -75.23 '-w ./ -W interferogram -m 20190104 --nofocus --zero --azimuth_looks 3 --range_looks 9 - f 0.6 -t 150 -b 250 -S TX
---------------------------
But when I do I get the following message and I have not been able to move forward:

****ops @ 4893514da1c7:** / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC $** stackStripMap.py -s / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / -d / home / work / geored-insar / Isce2 /Cartagena/DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -x '10 .06 10.36 -75.62 -75.23 '-w ./ -W interferogram -m 20190104 --nofocus --zero --azimuth_looks 3 - 9 - f 0.6 -t 150 -b 250 -S TX
/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py:5: DeprecationWarning: the imp module is deprecated in favor of importlib; see the module's documentation for alternative uses
  import os, imp, sys, glob
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed software at
https://winsar.unavco.org/software/isce
Sensor is not hard-coded (ers, envisat, alos), will keep default alks
['/ home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20190228', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/isce.log', ' / home / work / geored-insar / Isce2 / Cartagena / TS X / ASC / SLC / 20190104 ',' / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20200102 ',' / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20191130 ',' / home / work / geored-insa r / Isce2 / Cartagena / TSX / ASC / SLC / 20190322 ',' / home / work / geored -insar / Isce2 / Cartagena / TSX / ASC / SLC / 20190914 ',' / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20200308 ',' / h ome / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20200204 ',' / home / work / geored-insar / Isce2 / Cartagena / TSX / ASC / SLC / 20190607 ',' / home / work / geored-insar / Isce2 / Cartagena / TSX / AS C / SLC / 20191017 ']
['20190104', '20190228', '20190322', '20190607', '20190914', '20191017', '20191130', '20200102', '20200204', '20200308']
Traceback (most recent call last):
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 363, in <module>
    main ()
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 336, in main
    pairs = selectPairs (inps, stackReferenceDate, secondaryDates, acquisitionDates, doBaselines = True)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 745, in selectPairs
    baselineDict, timeDict = baselineStack (inps, stackReference, acuisitionDates, doBaselines)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 734, in baselineStack
    baselineDict [slv] = baselinePair (baselineDir, reference, secondary, doBaselines)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 705, in baselinePair
    bObj.wireInputPort (name = 'referenceFrame', object = mFrame)
  File "/opt/isce2/isce/components/iscesys/Component/Component.py", line 157, in method_with_flow
    return method (self) (* args, ** kwargs)
  File "/opt/isce2/isce/components/iscesys/Component/Component.py", line 259, in _wirePort
    raise PortError ("No% s port named% s"% (port_iterator.flow, name))
iscesys.Component.Component.PortError: 'No input port named referenceFrame'

I appreciate if you can help me solve this problem

Thanks

**Comment by fdelgadodelapuente** [comment]
Hi, 

the following command works well

stackStripMap.py -W interferogram -z --nofocus -s slcs -m 20121001 -d /data/francisco/yellowstone/dem/yellowstone_ned_isce.dem -t 800 -b 200 -a 8 -r 8 -u snaphu -f 0.5

**Comment by fgdiazm** [comment]
Hello Francisco,

Thanks for your suggestion but it didn't work, I tried with the option:

stackStripMap.py -W interferogram -z --nofocus -s /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/ -m 20190104 -d /home/work/geored-insar/Isce2/Cartagena/ DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -t 800 -b 200 -a 8 -r 8 -u snaphu -f 0.5

but i got the following error
/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py:5: DeprecationWarning: the imp module is deprecated in favor of importlib; see the module's documentation for alternative uses
  import os, imp, sys, glob
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed software at
https://winsar.unavco.org/software/isce
['/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190228', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/isce.log', ' /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190104', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200102', '/home/work /geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20191130', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190322', '/home/work/geored-insar /Isce2/Cartagena/TSX/ASC/SLC/20190914', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200308', '/home/work/geored-insar/Isce2/Cartagena /TSX/ASC/SLC/20200204', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190607', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC /SLC/20191017']
['20190104', '20190228', '20190322', '20190607', '20190914', '20191017', '20191130', '20200102', '20200204', '20200308']
Traceback (most recent call last):
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 363, in <module>
    main()
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 333, in main
    if inps.sensor.lower() == 'uavsar_stack': # don't try to calculate baselines for UAVSAR_STACK data
AttributeError: 'NoneType' object has no attribute 'lower'
***********************

I don't know if this message:

/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py:5: **DeprecationWarning: the imp module is deprecated in favor of importlib; see the module's documentation for alternative uses**

Requires some additional installation or configuration

**Reply by fdelgadodelapuente** [reply]
Change the following lines in 

    if inps.sensor and inps.sensor.lower() == 'uavsar_stack':    # don't try to calculate baselines for UAVSAR_STACK data
        pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=False)
    else:
        pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)

with 

    pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)

**Comment by fgdiazm** [comment]
now i tried with

stackStripMap.py -W interferogram -z --nofocus -s /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/ -m 20190104 -d /home/work/geored-insar/Isce2/Cartagena/ DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -t 800 -b 200 -a 8 -r 8 -u snaphu -f 0.5 -S TX


And in this case, I got the following message:

**/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py:5: DeprecationWarning: the imp module is deprecated in favor of importlib; see the module's documentation for alternative uses
  import os, imp, sys, glob**
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed software at
https://winsar.unavco.org/software/isce
Sensor is not hard-coded (ers, envisat, alos), will keep default alks
['/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190228', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/isce.log', ' /home/work/geored-insar/Isce2/Cartagena/T SX/ASC/SLC/20190104', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200102', '/home/ work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20191130', '/home/work/geored-in sar/Isce2/Cartagena/TSX/ASC/SLC/20190322', '/home/work/geored -insar/Isce2/Cartagena/TSX/ASC/SLC/20190914', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200308', '/home/work/geored-insar/Isce2 /Cartagena/TSX/ASC/SLC/20200204', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190607', '/home/work/geored-insar/Isce2/Cartagena/TS X/ASC/SLC/20191017']
['20190104', '20190228', '20190322', '20190607', '20190914', '20191017', '20191130', '20200102', '20200204', '20200308']
Traceback (most recent call last):
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 363, in <module>
    main()
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap.py", line 336, in main
    pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 745, in selectPairs
    baselineDict, timeDict = baselineStack(inps, stackReference, acuisitionDates,doBaselines)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 734, in baselineStack
    baselineDict[slv]=baselinePair(baselineDir, reference, secondary, doBaselines)
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/Stack.py", line 705, in baselinePair
    bObj.wireInputPort(name='referenceFrame', object=mFrame)
  File "/opt/isce2/isce/components/iscesys/Component/Component.py", line 157, in method_with_flow
    return method(self)(*args, **kwargs)
  File "/opt/isce2/isce/components/iscesys/Component/Component.py", line 259, in _wirePort
    raise PortError("No %s port named %s" % (port_iterator.flow, name))
iscesys.Component.Component.PortError: 'No input port named referenceFrame'
**********************************************

Is it the same case of the previous message?

**Reply by fdelgadodelapuente** [reply]
Don't add the -S flag. It is not necessary as you already specified the pixel ratio in the number of looks

**Comment by fgdiazm** [comment]
Hello Francisco,

I made a copy of stackStripMap.py and tried to change the line in the following ways:

one)
if inps.sensor.lower() == 'uavsar_stack': # don't try to calculate baselines for UAVSAR_STACK data
    pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)
  else:
    pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)
  print ('number of pairs: ', len(pairs))

**but i get the following error**

ops@ebedd7272f41:/opt/isce2/isce2/contrib/stack/stripmapStack$ stackStripMap_modtry2.py -W interferogram -z --nofocus -s /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/ - m 20190104 -d /home/work/geored-insar/Isce2/Cartagena/DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -t 800 -b 200 -a 8 -r 8 -u snaphu -f 0.5
/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap_modtry2.py:5: DeprecationWarning: the imp module is deprecated in favor of importlib; see the module's documentation for alternative uses
  import os, imp, sys, glob
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed software at
https://winsar.unavco.org/software/isce
['/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190228', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/isce.log', ' /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190104', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200102', '/home/work /geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20191130', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190322', '/home/work/geored-insar /Isce2/Cartagena/TSX/ASC/SLC/20190914', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20200308', '/home/work/geored-insar/Isce2/Cartagena /TSX/ASC/SLC/20200204', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/20190607', '/home/work/geored-insar/Isce2/Cartagena/TSX/ASC /SLC/20191017']
['20190104', '20190228', '20190322', '20190607', '20190914', '20191017', '20191130', '20200102', '20200204', '20200308']
Traceback (most recent call last):
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap_modtry2.py", line 363, in <module>
    main()
  File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap_modtry2.py", line 333, in main
    if inps.sensor.lower() == 'uavsar_stack': # don't try to calculate baselines for UAVSAR_STACK data
AttributeError: 'NoneType' object has no attribute 'lower'
************************************************** ***********
**Sorry if I didn't understand**

**Comment by fgdiazm** [comment]
**On the other hand I deleted**

if inps.sensor and inps.sensor.lower() == 'uavsar_stack': # don't try to calculate baselines for UAVSAR_STACK data
     pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=False)
else:
     pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)

**and I just left**

pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)

**and it looks like this:**

stackStripMap_mod20220117.py -W interferogram -z --nofocus -s /home/work/geored-insar/Isce2/Cartagena/TSX/ASC/SLC/ -m 20190104 -d /home/work/geored-insar/Isce2/Cartagena/ DEM/demLat_N09_N11_Lon_W077_W074.dem.wgs84 -t 800 -b 200 -a 8 -r 8 -u snaphu -f 0.5
   File "/opt/isce2/isce2/contrib/stack/stripmapStack/stackStripMap_mod20220117.py", line 333
     pairs = selectPairs(inps,stackReferenceDate, secondaryDates, acquisitionDates,doBaselines=True)
     ^
IndentationError: unexpected indent

**Comment by volter29** [comment]
HI @fgdiazm 
inside the SLC folder you will find a series of data called raw.bak raw.dat raw.dir. Once deleted, restart stackStripmap
